src/main/java/homework_4/singleton/Singleton.java

```diff
@@ -0,0 +1,28 @@
+package homework_4.singleton;
+
+public class Singleton {
+
+    private static volatile Singleton INSTANCE;
+    private static String nameOfInstance;
+
+    private Singleton(String nameOfInstance) {
+        Singleton.nameOfInstance = nameOfInstance;
+    }
+
+    public static Singleton getInstance(String nameOfInstance) {
+        Singleton local = INSTANCE;
+        if (local == null) {
+            synchronized (Singleton.class) {
+                local = INSTANCE;
+                if (local == null) {
+                    INSTANCE = local = new Singleton(nameOfInstance);
+                }
+            }
+        }
+        return local;
```

Comment:
Opt: it seems there are some unnecessary branching and code conventions are not followed

Reply:
I see you used Double Checked Locking, then I hope you really understood what you're doing and not just copypasted it)